### PR TITLE
[ENH] Use standard T2* map as coregistration target

### PR DESCRIPTION
**Pull request by emdupre**
## Changes proposed in this pull request

For MEEPI, changes the T2* map target from the adaptive to the standard T2* map. This updates the coregistration target in the case of `--t2s-coreg` and seems to produce more consistent alignment.

I have also updated the docstring for the `t2s_coreg` argument in all relevant workflows and updated the workflow description for `bold_t2s_wf`.

## Documentation that should be reviewed

The workflow description will need review to ensure its correctness.

### CODE REVIEW

File: fmriprep/workflows/base.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def init_fmriprep_wf(subject_list, task_id, echo_idx, run_uuid, work_dir, output
             Treat multiple sessions as longitudinal (may increase runtime)
             See sub-workflows for specific differences
         t2s_coreg : bool
-            Use multiple BOLD echos to create T2*-map for T2*-driven coregistration
+            For multi-echo EPI, used the calculated T2*-map for T2*-driven coregistration
```
Comment (effigies) [inline fmriprep/workflows/base.py]:
```suggestion
            For multi-echo EPI, use the calculated T2*-map for T2*-driven coregistration
```

File: fmriprep/workflows/base.py [hunk 2]
```diff
@@ -309,7 +309,7 @@ def init_single_subject_wf(subject_id, task_id, echo_idx, name, reportlets_dir,
             Treat multiple sessions as longitudinal (may increase runtime)
             See sub-workflows for specific differences
         t2s_coreg : bool
-            Use multiple BOLDS echos to create T2*-map for T2*-driven coregistration
+            For multi-echo EPI, used the calculated T2*-map for T2*-driven coregistration
```
Comment (effigies) [inline fmriprep/workflows/base.py]:
```suggestion
            For multi-echo EPI, use the calculated T2*-map for T2*-driven coregistration
```

File: fmriprep/workflows/bold/t2s.py [hunk 3]
```diff
@@ -67,14 +68,14 @@ def init_bold_t2s_wf(echo_times,
...
+A T2* map was estimated from the preprocessed BOLD by fitting to a monoexponential signal
+decay model with log-linear regression.
+For each voxel, the maximal number of echoes with reliable signal in that voxel were
 used to fit the model.
-The T2* map was used to optimally combine preprocessed BOLD across
-echoes following the method described in @posse_t2s and was also retained as
-the BOLD reference.
-"""
+The calculated T2* map was then used to optimally combine preprocessed BOLD across
+echoes following the method described in @posse_t2s. The optimally combined time
```
Comment (effigies) [inline fmriprep/workflows/bold/t2s.py]:
```suggestion
echoes following the method described in @posse_t2s.
The optimally combined time series was carried forward as the *preprocessed BOLD*{}.
```

File: fmriprep/workflows/bold/t2s.py [hunk 4]
```diff
@@ -67,14 +68,14 @@ def init_bold_t2s_wf(echo_times,
...
+series was carried forward as the *preprocessed BOLD*{}.
```
Comment (effigies) [inline fmriprep/workflows/bold/t2s.py]:
```suggestion
```

File: fmriprep/workflows/bold/t2s.py [hunk 5]
```diff
@@ -67,14 +68,14 @@ def init_bold_t2s_wf(echo_times,
...
+""".format('.' if not t2s_coreg else ', and the T2* map was also retained as the BOLD reference')
```
Comment (effigies) [inline fmriprep/workflows/bold/t2s.py]:
```suggestion
""".format('' if not t2s_coreg else ', and the T2* map was also retained as the BOLD reference')
```

File: fmriprep/workflows/bold/t2s.py [hunk 6]
```diff
@@ -67,14 +68,14 @@ def init_bold_t2s_wf(echo_times,
...
+A T2* map was estimated from the preprocessed BOLD by fitting to a monoexponential signal
+decay model with log-linear regression.
+For each voxel, the maximal number of echoes with reliable signal in that voxel were
 used to fit the model.
-The T2* map was used to optimally combine preprocessed BOLD across
-echoes following the method described in @posse_t2s and was also retained as
-the BOLD reference.
-"""
+The calculated T2* map was then used to optimally combine preprocessed BOLD across
+echoes following the method described in @posse_t2s.
```
Comment (oesteban) [inline fmriprep/workflows/bold/t2s.py]:
```suggestion
echoes following the method described in [@posse_t2s].
```

File: fmriprep/workflows/bold/t2s.py [hunk 7]
```diff
@@ -67,14 +68,14 @@ def init_bold_t2s_wf(echo_times,
...
-For each voxel, the maximal number of echoes with high signal in that voxel was
+A T2* map was estimated from the preprocessed BOLD by fitting to a monoexponential signal
+decay model with log-linear regression.
+For each voxel, the maximal number of echoes with reliable signal in that voxel were
 used to fit the model.
-The T2* map was used to optimally combine preprocessed BOLD across
-echoes following the method described in @posse_t2s and was also retained as
-the BOLD reference.
-"""
+The calculated T2* map was then used to optimally combine preprocessed BOLD across
```
Comment (oesteban) [inline fmriprep/workflows/bold/t2s.py]: What does this mean? (this is not a question about the PR, it is to fill a gap of ignorance on my side, and potentially ask for a little amend for people of my level of knowledge about MEEPI)

File: fmriprep/workflows/bold/t2s.py [hunk 8]
```diff
@@ -89,7 +90,7 @@ def init_bold_t2s_wf(echo_times,
     workflow.connect([
         (inputnode, t2smap_node, [('bold_file', 'in_files')]),
         (t2smap_node, outputnode, [('optimal_comb', 'bold')]),
-        (t2smap_node, skullstrip_t2smap_wf, [('t2star_adaptive_map', 'inputnode.in_file')]),
+        (t2smap_node, skullstrip_t2smap_wf, [('t2star_map', 'inputnode.in_file')]),
```
Comment (oesteban) [inline fmriprep/workflows/bold/t2s.py]: shouldn't this be:
```suggestion
        (t2smap_node, skullstrip_t2smap_wf, [('t2s_vol', 'inputnode.in_file')]),
```

as per https://github.com/poldracklab/fmriprep/blob/a783c38c72cceadaf4ef0ddb6abfee947b184a2f/fmriprep/interfaces/multiecho.py#L129?

Comment (emdupre) [reply fmriprep/workflows/bold/t2s.py]: This got updated in #1296 ! Now it's: https://github.com/poldracklab/fmriprep/blob/1c25175980522ab037366b3055797408b4709fe9/fmriprep/interfaces/multiecho.py#L45

We also return quite a few more outputs, now !

File: fmriprep/workflows/bold/t2s.py (hunk 6 above)
Comment (emdupre) [reply fmriprep/workflows/bold/t2s.py]: Great catch !

File: fmriprep/workflows/bold/t2s.py (hunk 7 above)
Comment (emdupre) [reply fmriprep/workflows/bold/t2s.py]: We use the T2* map to determine the weightings given to each echo in the optimal combination. Should I update the text here, to reference that ? It's explained in the reference, but I can also put more information, here !

Comment (oesteban) [reply fmriprep/workflows/bold/t2s.py]: I think it is okay, let's not delay this PR :) Thanks!

**Pull request merged**